### PR TITLE
RUMM-88 Add provider for first party RUM Resource/Error events

### PR DESCRIPTION
**Pull request by mariusc83**
…vents

### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -6,15 +6,6 @@
           <package name="kotlinx.android.synthetic" alias="false" withSubpackages="true" />
         </value>
       </option>
-      <option name="PACKAGES_IMPORT_LAYOUT">
```
Comment (xgouchet) [inline .idea/codeStyles/Project.xml:9]: Is this coming from the IntelliJ update?

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt [hunk 2]
```diff
@@ -180,6 +178,14 @@ internal class RumResourceScope(
         sent = true
     }
 
+    private fun resolveResourceProvider(): ResourceEvent.Provider? {
+        return if (firstPartyHostDetector.isFirstPartyUrl(url)) {
+            ResourceEvent.Provider(domain = url, type = ResourceEvent.Type2.FIRST_PARTY)
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt:183]: I think the domain is meant to be the host. You can look at how this value is filled to make sure.

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (mariusc83) [reply .idea/codeStyles/Project.xml:9]: Yup...I just switched to AS

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt (hunk 2 above)
Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt]: Yup. I changed that while you were reviewing this :).

**Review by xgouchet — APPROVED** [review]

**Pull request merged**